File: frontend/src/dto/Visit.ts
import {Doctor} from "./Doctor";

export interface Visit {
    id?: number;
    lastName?: string;
    startHour?: number;
    firstName?: string;
    middleName?: string;
    dateOfReceipt?: String;
    doctor?: Doctor;
    doctorId?: number;

    patientId?: number
    isVisited?: boolean;
}

File: backend/src/main/java/com/example/backend/entity/Visit.java
package com.example.backend.entity;

import java.time.*;

import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Visit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;

    private LocalDate dateOfReceipt;

    private int startHour;

    private Boolean isVisited = false;

    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;

    @ManyToOne
    @JoinColumn(name = "doctor_id")
    private Doctor doctor;

}
